lib/daru/vector.rb

```diff
@@ -199,19 +199,62 @@ def initialize source, opts={}
     #   # For vectors employing hierarchial multi index
     #
     def [](*input_indexes)
-      # Get a proper index object
-      indexes = @index[*input_indexes]
+      # Get array of positions indexes
+      positions = @index.pos(*input_indexes)
```

Comment:
If I am not missing something, from this place down you can just replace everything with `at(*positions)` and call it a day.